File: backend/src/main/java/com/snackoverflow/toolgether/domain/user/dto/MyReservationInfoResponse.java
package com.snackoverflow.toolgether.domain.user.dto;

import com.snackoverflow.toolgether.domain.reservation.entity.Reservation;
import lombok.NonNull;

import java.time.LocalDateTime;

public record MyReservationInfoResponse(
        @NonNull Long id,
        @NonNull String title,
        String image,
        @NonNull Double amount,
        @NonNull LocalDateTime startTime,
        @NonNull LocalDateTime endTime,
        @NonNull String status,
        @NonNull Boolean isReviewed
) {
    public static MyReservationInfoResponse from(Reservation reservation, String imageUrl, boolean isReviewed) {

        return new MyReservationInfoResponse(
                reservation.getId(),
                reservation.getPost().getTitle(),
                imageUrl,
                reservation.getAmount(),
                reservation.getStartTime(),
                reservation.getEndTime(),
                reservation.getStatus().toString(),
                isReviewed
        );
    }
}

File: backend/src/main/java/com/snackoverflow/toolgether/domain/user/dto/response/MyReservationInfoResponse.kt
package com.snackoverflow.toolgether.domain.user.dto.response;

import com.snackoverflow.toolgether.domain.reservation.entity.Reservation;
import kotlin.jvm.JvmStatic;

import java.time.LocalDateTime;

data class MyReservationInfoResponse(
    val id: Long?,
    val title: String,
    val image: String?,
    val amount: Double,
    val startTime: LocalDateTime,
    val endTime: LocalDateTime,
    val status: String,
    val isReviewed: Boolean
) {
    companion object {
        @JvmStatic
        fun from(reservation: Reservation, imageUrl: String?, isReviewed: Boolean): MyReservationInfoResponse {
            return MyReservationInfoResponse(
                    id = reservation.id,
                    title = reservation.post.title,
                    image = imageUrl,
                    amount = reservation.amount,
                    startTime = reservation.startTime,
                    endTime = reservation.endTime,
                    status = reservation.status.toString(),
                    isReviewed = isReviewed
            )
        }
    }
}